services/controllerhost/queueDepth.go

```diff
@@ -544,10 +545,17 @@ func (qdc *queueDepthCalculator) handleStartFrom(
 		if qualify {
 			trace += 100
 			consumerGroupExtent.WriteTime = common.Int64Ptr(int64(now))
-			consumerGroupExtent.AckLevelSeqNo = common.Int64Ptr(common.MaxInt64(
-				storeMetadata.beginSequence, // Retention may have removed some messages
-				int64(startFromSeq),         // Otherwise, act like we had just opened this extent at startFrom, i.e. don't count messages before startFrom
-			))
+
+			// retention may have purged some messages, account for that
+			if storeMetadata.beginSequence != math.MaxInt64 {
```

Comment:
(1) this changes behavior, you still want to update AckLevelSeqNo if quality=true
(2) trace logging is broken
(3) Needs to be backwards compatible (i.e. also deal with beginSeq=-1)